vermouth/processors/do_mapping.py

```diff
@@ -445,12 +450,38 @@ def apply_mod_mapping(match, molecule, graph_out, mol_to_out, out_to_mol):
                 raise ValueError("No node found in molecule with "
                                  "atomname {}".format(modification.nodes[mod_idx]['atomname']))
             # Undefined loop variable is guarded against by the else-raise above
+            anchors.add(mod_idx)
             mod_to_out[mod_idx] = out_idx  # pylint: disable=undefined-loop-variable
             graph_out.nodes[out_idx].update(modification.nodes[mod_idx].get('replace', {})) # pylint: disable=undefined-loop-variable
         graph_out.nodes[out_idx]['modifications'] = graph_out.nodes[out_idx].get('modifications', [])
         if modification not in graph_out.nodes[out_idx]['modifications']:
             graph_out.nodes[out_idx]['modifications'].append(modification)
 
+    # FIXME Jank here to ensure the charge_group attributes look reasonable
+    charge_group_start = max(graph_out.nodes[mod_to_out[idx]].get('charge_group', 1) for idx in anchors) if anchors else 1
+    for charge_group, mod_idx in enumerate(modification, charge_group_start):
+        out_idx = mod_to_out[mod_idx]
+        if 'charge_group' not in graph_out.nodes[out_idx]:
+            graph_out.nodes[out_idx]['charge_group'] = charge_group
+
+    # FIXME we need to assign updated resids to the modification atoms
+    # the block mapping does this for the unmodified atoms
+    for mod_idx, out_idx in mod_to_out.items():
+        if mod_idx in anchors:
+            continue
+        neighbors = modification.neighbors(mod_idx)
+        resid_new = None
+        for mod_idx_neigh in neighbors:
+            anchor = mod_to_out[mod_idx_neigh]
+            resid_new = graph_out.nodes[anchor].get('resid', None)
+            resid_old = graph_out.nodes[anchor].get('_old_resid', None)
+            if resid_new:
+                graph_out.nodes[out_idx]['_old_resid'] = resid_old
+                graph_out.nodes[out_idx]['resid'] = resid_new
+                break
+        if not resid_new:  # Last found resid is *still* None....
+            raise ValueError(f"No resid found for {format_atom_string(graph_out.nodes[out_idx])}")
```

Comment:
I'm contemplating whether this, or at least the second half, should be moved to a separate processor.